### PR TITLE
KEYCLOAK-12867: Credential title should be visible if cred type disabled

### PR DESCRIPTION
**Pull request by ssilvert**

### CODE REVIEW

**Comment by ssilvert** [comment]
@mposolda This fix uncovered a bug in the REST API.  See https://issues.redhat.com/browse/KEYCLOAK-12869

File: themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx
Comment (vmuzikar) [inline themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx]: Should we really use user label as a key? It can contain special character, white spaces, etc.
Or is that ok? AFAIK `key` doesn't appear anywhere in the final HTML, right?

**Comment by mposolda** [comment]
@ssilvert Looks good to me, Thanks.

I will re-trigger travis (it failed due temporarily broken Keycloak master, which is unrelated to this PR). If Travis pass, I will approve.

File: themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx
Comment (mposolda) [reply themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx]: No clue if it is safe to use userLabel as a key here or not :)

Just a note that userLabel can be null or empty. And yes, it can contain spaces and national characters with diacritics etc.

Comment (ssilvert) [reply themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx]: In line 200 I make sure that userLabel will have a value.  If you don't send me a value for userLabel then it gets set to Msg.localize(credential.type).

The key does not appear on the screen but it does appear in the HTML.  It can be any string.  It just needs to be unique to its siblings in a loop.  This is used by React to identify each item when re-rendering.  It's really not very important for this app, but I include them to keep React from putting a warning in the javascript console.

Come to think of it, I have the ID, so that's really a better choice than userLabel.  I'll fix that soon, but not in this PR.

Comment (vmuzikar) [reply themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx]: +1 for using ID which is clearly a better choice for stuff like that.

@ssilvert Could you please create a JIRA for this so we don't forget?

**Comment by mposolda** [comment]
@ssilvert @vmuzikar I don't have more comments to this PR and leaving to you guys whether to address userLabel comment in this PR or somewhere else. I will approve once @vmuzikar approves

File: themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx
Comment (ssilvert) [reply themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx]: I went ahead and fixed it then squashed commits.

Comment (vmuzikar) [reply themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx]: Thanks!

**Review by vmuzikar — APPROVED** [review]
LGTM.

**Review by mposolda — APPROVED** [review]
@ssilvert @vmuzikar Thanks!

**Pull request merged**
